aiida/tools/graph/graph_traversers.py

```diff
@@ -8,34 +8,59 @@
 # For further information please visit http://www.aiida.net               #
 ###########################################################################
 """Module for functions to traverse AiiDA graphs."""
+import sys
+from typing import Any, Callable, cast, Dict, Iterable, List, Mapping, Optional, Set
 
 from numpy import inf
-from aiida.common.links import GraphTraversalRules, LinkType
-
 
-def get_nodes_delete(starting_pks, get_links=False, **kwargs):
+from aiida import orm
+from aiida.common import exceptions
+from aiida.common.links import GraphTraversalRules, LinkType
+from aiida.orm.utils.links import LinkQuadruple
+from aiida.tools.graph.age_entities import Basket
+from aiida.tools.graph.age_rules import UpdateRule, RuleSequence, RuleSaveWalkers, RuleSetWalkers
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+
+    class TraverseGraphOutput(TypedDict, total=False):
+        nodes: Set[int]
+        links: Optional[Set[LinkQuadruple]]
+        rules: Dict[str, bool]
+else:
+    TraverseGraphOutput = Mapping[str, Any]
+
+
+def get_nodes_delete(
+    starting_pks: Iterable[int],
+    get_links: bool = False,
+    missing_callback: Optional[Callable[[Iterable[int]], None]] = None,
+    **traversal_rules: bool
+) -> TraverseGraphOutput:
     """
     This function will return the set of all nodes that can be connected
     to a list of initial nodes through any sequence of specified authorized
     links and directions for deletion.
 
-    :type starting_pks: list or tuple or set
     :param starting_pks: Contains the (valid) pks of the starting nodes.
 
-    :param bool get_links:
+    :param get_links:
         Pass True to also return the links between all nodes (found + initial).
 
-    :param bool create_forward: will traverse CREATE links in the forward direction.
-    :param bool call_calc_forward: will traverse CALL_CALC links in the forward direction.
-    :param bool call_work_forward: will traverse CALL_WORK links in the forward direction.
+    :param missing_callback: A callback to handle missing starting_pks or if None raise NotExistent
```

Comment:
might be useful to add your example implementation here

Reply:
something like this?

```suggestion
    :param missing_callback: A callback to handle missing starting_pks or if None raise NotExistent
    	For example to ignore them: ``missing_callback=lambda missing_pks: None``
```